Frameworks/Foundation/NSString.mm

```diff
@@ -537,6 +537,10 @@ - (BOOL)getBytes:(void*)buffer
  @Status Interoperable
 */
 - (void)getCharacters:(unichar*)dest range:(NSRange)range {
+    if (range.location + range.length > self.length) {
```

Comment:
surprised we didn't have this check already. use NSLocationInRange instead, though?